iron-doc-viewer.html

```diff
@@ -195,6 +211,34 @@
       this._behaviors  = this.descriptor.behaviors || [];
 
       this.toggleAttribute('abstract', this.descriptor.abstract);
+
+      // On initial load, force position to the correct anchor.
+      // (This shouldn't be required in the catalog, but the catalog
+      // may scroll before content is fully loaded, so we try again here.)
+      this.scrollToAnchor();
+    },
+
+    /**
+     * Scrolls to the currently selected anchor, as identified
+     * by the URL hash. Must called from the outside if the
+     * hash changes but the selected element doesn't (for example,
+     * the user clicks on a property name on the current page).
+     */
+    scrollToAnchor: function() {
+      // scroll to anchor after data finishes loading
+      if (window.location.hash) {
+        // ToDo: this async appears to be required
+        // on catalog, but it's unclear why.
+        this.async(function() {
```

Comment:
this is similar to the work @ebidel did in iron-component-page: https://github.com/PolymerElements/iron-component-page/commit/2a07cd9b28a237ae79e60170c3e547af6f568414, so we may need only one.